ExampleMod/Content/Items/Weapons/ExampleCustomUseStyleWeapon.cs

```diff
@@ -0,0 +1,205 @@
+			}
+
+			// Here we set the front arm drawing parameters. This uses the newer arm rendering approach.
+			// The normal vanilla Swing doesn't use this approach and instead only uses the old approach of setting player.bodyFrame.Y during UseItemFrame.
+			// The old approach has discrete arm positions while the new approach can draw the arm at any angle.
+			player.SetCompositeArmFront(true, Player.CompositeArmStretchAmount.Full, currentAngle - MathHelper.PiOver2);
+			// We could also use SetCompositeArmBack to set the drawing parameters of the other arm.
+
+			// We set itemLocation to indicate where the item should be drawn
+			player.itemLocation = player.MountedCenter + currentAngle.ToRotationVector2() * (HandOffsets.TryGetValue(item.type, out int offset) ? new Vector2(offset) : Vector2.Zero);
+
+			// This FlipItemLocationAndRotationForGravity method handles adjusting itemRotation and itemLocation to account for reversed gravity.
+			player.FlipItemLocationAndRotationForGravity();
+			/* This is what the method does:
+			if (player.gravDir == -1f) {
+				player.itemRotation = 0f - player.itemRotation;
+				player.itemLocation.Y = player.position.Y + (float)player.height + (player.position.Y - player.itemLocation.Y);
+			}
+			*/
+		}
+
+		// We use UseItemHitbox to determine the hitbox of the item during the weapon animation
+		public override void UseItemHitbox(Item item, Player player, ref Rectangle hitbox, ref bool noHitbox) {
+			// Helper method to sort coordinates later
+			(int, int) Order(float v1, float v2) {
+				if (v1 < v2) {
+					return ((int)v1, (int)v2);
+				}
+
+				return ((int)v2, (int)v1);
+			}
+
+			// Calculate the direction of the hand
+			Vector2 handDirection = (player.compositeFrontArm.rotation + MathHelper.PiOver2).ToRotationVector2() * player.gravDir;
+
+			// Calculate the item hitbox. The hitbox parameter passed in already has item scale applied so recalculating it here is more flexible.
+			Rectangle drawHitbox = Item.GetDrawHitbox(item.type, player);
+			Vector2 itemSize = Main.dedServ ? new Vector2(32) : new Vector2(drawHitbox.Width, drawHitbox.Height);
+
+			// Calculate the distance from the handle to the tip of the weapon, taking into account item scaling effects.
+			float itemLength = (itemSize * player.GetAdjustedItemScale(item)).Length(); // We don't use Item.Size/width/height because those values are for the item hitbox when dropped.
+
+			// Calculate the handle and tip positions
+			Vector2 handlePosition = handDirection * (HandOffsets.TryGetValue(item.type, out int offset) ? new Vector2(offset) : Vector2.Zero) + player.MountedCenter;
+			Vector2 tipPosition = handlePosition + handDirection * itemLength;
+
+			// Now we use those values to create the item hitbox
+			(int X1, int X2) = Order(handlePosition.X, tipPosition.X);
+			(int Y1, int Y2) = Order(handlePosition.Y, tipPosition.Y);
+			hitbox = new Rectangle(X1, Y1, X2 - X1, Y2 - Y1);
```

Comment:
This is kinda sad, would be nice if we had `Utils.CornerRect`, or `BoundingRect(p1, p2)`

In 1.4.5 I do it like this:
```cs
			var r = Utils.CenteredRectangle(points[0], Vector2.Zero);
			foreach (var point in points)
				r = r.Including(point.ToPoint());

			BoundingRect = r;
```
			
Where `Including` is an extension method in `Utils`
```cs
		public static Rectangle Including(this Rectangle rect, Point point)
		{
			int l = Math.Min(rect.Left, point.X);
			int r = Math.Max(rect.Right, point.X);
			int t = Math.Min(rect.Top, point.Y);
			int b = Math.Max(rect.Bottom, point.Y);
			return new Rectangle(l, t, r - l, b - t);
		}
```

Reply:
Added.